src/main/java/com/google/devtools/build/lib/skylarkbuildapi/ProtoInfoApi.java

```diff
@@ -37,7 +37,7 @@
 public interface ProtoInfoApi<FileT extends FileApi> extends StructApi {
   /** Provider class for {@link ProtoInfoApi} objects. */
   @SkylarkModule(name = "Provider", documented = false, doc = "")
-  interface Provider extends ProviderApi {
+  interface ProtoInfoProviderApi extends ProviderApi {
```

Comment:
Same here -- why is `Provider` not adequate? It's shorter so all other things being equal, it's better.

Reply:
Referring to `ProtoInfoProviderApi` instead of `ProtoInfoApi.Provider` seems a little bit cleaner and it's the same pattern used by `JavaInfoProviderApi`